cirq/study/sweeps.py

```diff
@@ -101,6 +102,38 @@ def __iter__(self) -> Iterator[resolver.ParamResolver]:
         for params in self.param_tuples():
             yield resolver.ParamResolver(collections.OrderedDict(params))
 
+    # pylint: disable=function-redefined
+    @overload
+    def __getitem__(self, val: int) -> resolver.ParamResolver:
+        pass
+
+    # pylint: disable=function-redefined
+    @overload
+    def __getitem__(self, val: slice) -> 'Sweep':
+        pass
+
+    def __getitem__(self, val):
+        if isinstance(val, int):
+            if val < 0:
+                val += len(self)
+            val = slice(val, val + 1, None)
```

Comment:
can just do
```python
return next(itertools.islice(self, val))
```

Reply:
Actually, that suggestion is also not quite right. Imagine sweep were length 10 and you asked for `sweep[-20]`; the code above would return the 0th element, but should raise an `IndexError`. I think we need to a bounds check:
```python
if isinstance(val, int):
    if val < -len(self) or val >= len(self):
        raise IndexError(f'sweep index out of range: {val}')
    if val < 0:
        val += len(self)
    return next(itertools.islice(self, val))
```
It might be nice to make a local variable `n = len(self)` at the top of the function to avoid calling `len(self)` multiple times.

Reply:
changed.

Reply:
Looks like the fix does not reflect my follow-up comment about bounds checking. Should fix that to ensure we raise `IndexError` for out-of-bounds indexes, instead of returning the zeroth element or raising `StopIteration`.